### PR TITLE
WIP Disable in-tree plugins when CSI Migration enabled

### PR DESCRIPTION
**Pull request by ddebroy**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test

/kind feature

> /kind flake

**What this PR does / why we need it**:
This PR disables the main entry points of the in-tree plugins whose code is being migrated to CSI. This paves the path to provide us more confidence that the CSI migration shims are indeed working and the in-tree code, after being disabled for a while (with feature flags) can be removed.

**Which issue(s) this PR fixes**:
Fixes # https://github.com/kubernetes/kubernetes/issues/77237

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Disable in-tree plugins when corresponding feature flags for CSI migration are enabled.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

### CODE REVIEW

File: cmd/kube-controller-manager/app/plugins.go [hunk 1]
```diff
@@ -65,13 +65,18 @@ import (
 // initialization code for kubelet really, really need a through refactor.
 func ProbeAttachableVolumePlugins() []volume.VolumePlugin {
 	allPlugins := []volume.VolumePlugin{}
-
-	allPlugins = append(allPlugins, awsebs.ProbeVolumePlugins()...)
-	allPlugins = append(allPlugins, gcepd.ProbeVolumePlugins()...)
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigration) || !utilfeature.DefaultFeatureGate.Enabled(features.CSIMigrationAWS) {
```
Comment (davidz627) [inline cmd/kube-controller-manager/app/plugins.go]: could a minor refactor be done so we don't need to duplicate this feature gate checking code? Looks like all these various `Probe` functions have generally the same code and the same set of plugins. Wonder if they could be consolidated for the most part

Comment (ddebroy) [reply cmd/kube-controller-manager/app/plugins.go]: Sure. We can discuss a bit if we want to align this more with https://github.com/kubernetes/kubernetes/pull/80353/files#diff-0967aee5156ab01aec200f53706a7135. I think for now, I can group them under `append*LegacyProviderVolumes` like @BenTheElder did.

**Comment by ddebroy** [comment]
The PR test failure in pull-kubernetes-integration :: `TestVolumeProvision` is due to https://github.com/kubernetes/kubernetes/issues/80528

**Comment by ddebroy** [comment]
Needs to properly handle resizing: FindExpandablePluginBySpec will not return a plugin that is not registered/removed.

**Comment by davidz627** [comment]
I believe this PR has been replaced by https://github.com/kubernetes/kubernetes/pull/83098
/close

**Pull request closed without merging**